### ISSUE TITLE
Crash when turning off advanced slider on support tab

### ISSUE THREAD

**Issue by kjcolley7** [SoftFever/OrcaSlicer#6519]
### Is there an existing issue for this problem?

- [X] I have searched the existing issues

### OrcaSlicer Version

2.1.1

### Operating System (OS)

macOS

### OS Version

macOS Sonoma 14.6.1 (23G93)

### Additional system information

Apple M1 Pro, 32GB RAM

### Printer

Ender 3 V2 (running Klipper)

### How to reproduce

I haven't actually tried reproducing, but this is approximately what I did:

1. Have the advanced slider on
2. Go to the support tab
3. Turn the advanced slider off
4. Instant crash

### Actual results

Orca Slicer crashed with this as the report body:

[OrcaSlicer.crash.zip](https://github.com/user-attachments/files/16654876/OrcaSlicer.crash.zip)

### Expected results

No crash

### Project file & Debug log uploads

[debug_Sun_Aug_18_18_51_17_11928.log.0.zip](https://github.com/user-attachments/files/16654795/debug_Sun_Aug_18_18_51_17_11928.log.0.zip)

### Checklist of files to include

- [X] Log file
- [ ] Project file

**Comment by Noisyfox** [comment]
Have you tried the latest nightly build?

**Comment by Noisyfox** [comment]
I wonder if this is the same issue:
![b412bc21a3c8af4288320806ac25bf47](https://github.com/user-attachments/assets/6025d9fd-df1d-44b7-83fc-891d1122f7a3)

**Comment by Noisyfox** [comment]
Could you try the build from this PR #6551 and see if that solves the problem?

**Comment by kjcolley7** [comment]
I also haven't been able to reproduce the issue by doing the same steps as before. Seeing as the crash log shows that it is a heap allocation lifetime issue, it might be a sporadic crash that's tricky to reproduce. Since this crash has only happened once, it'll be tricky to see if a new build fixes the problem.

**Comment by kjcolley7** [comment]
I reproduced the crash by switching to the machine g-code tab in printer settings, closing the window, and then toggling the advanced slider off (as shown in the GIF in PR #6551). This is the crash I got, which looks like it matches the crash I'm reporting in this issue:

[OrcaSlicer2.crash.zip](https://github.com/user-attachments/files/16738103/OrcaSlicer2.crash.zip)

Therefore, I assume this issue will be fixed by commit 934f171b836e5c79a30ac03a4310d58bbd380f37, once it merges. Thanks for investigating this, especially considering the lack of information I was able to provide in the issue!
